src/ast/grouping.rs

```diff
@@ -0,0 +1,76 @@
+use crate::ast::{Column, DatabaseValue};
+
+pub type GroupByDefinition = (DatabaseValue);
+
+/// A list of definitions for the `GROUP BY` statement
+#[derive(Debug, Default, PartialEq, Clone)]
+pub struct Grouping(pub Vec<GroupByDefinition>);
+
+impl Grouping {
+    #[doc(hidden)]
+    pub fn append(mut self, value: GroupByDefinition) -> Self {
+        self.0.push(value);
+        self
+    }
+
+    #[inline]
+    pub fn new(values: Vec<GroupByDefinition>) -> Self {
+        Self(values)
+    }
+
+    #[inline]
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+}
+
+/// An item that can be used in the `GROUP BY` statement
+pub trait Groupable
+where
+    Self: Sized,
+{
+    /// Group by `self`
+    fn group(self) -> GroupByDefinition;
+}
+
+/// Convert the value into a group by definition.
+pub trait IntoGroupByDefinition {
```

Comment:
Why not using `From`?

```rust
impl<'a> From<&'a str> for GroupByDefinition {
    fn from(self) -> GroupByDefinition { ... }
}
```

Reply:
I was just replicating the pattern used for `IntoOrderDefinition`. Should I change to `From`?

Reply:
No, leave it how it is. I remember now there was some limitations in place here why we needed our own trait.